--- /usr/src/aolserver45/xotcl-1.5.3/library/serialize/Serializer.xotcl	2006-09-27 10:12:40.000000000 +0200
+++ serialize/Serializer.xotcl	2007-06-25 12:39:32.000000000 +0200
@@ -331,11 +331,27 @@
     #if {[my needsOneOf [$x info instmixin ]]} {return 0}
     return 1
   }
+  Serializer instproc all-slots {obj} {
+    set type [$obj info class]
+    set slot_objs [list]
+     foreach cl [concat $type [$type info heritage]] {
+      foreach s [$cl info slots] { 
+       if {![info exists check([namespace tail $s])]} {
+	 set check([namespace tail $s]) 1
+	 lappend slot_objs $s
+       }
+     }
+   }
+   
+  return $slot_objs
+
+  }
   Serializer instproc Object-needsNothing x {
     set p [$x info parent]
     if {$p ne "::"  && [my needsOneOf $p]} {return 0}
     if {[my needsOneOf [$x info class]]}  {return 0}
-    if {[my needsOneOf [[$x info class] info slots]]}  {return 0}
+    #if {[my needsOneOf [[$x info class] info slots]]}  {return 0}
+    if {[my needsOneOf [my all-slots $x]]}  {return 0} 
     #if {[my needsOneOf [$x info mixin ]]} {return 0}
     return 1
   }
